integrations/access/pagerduty/testlib/suite.go

```diff
@@ -453,24 +462,32 @@ func (s *PagerdutySuiteOSS) TestRecipientsFromAccessMonitoringRule() {
 		})
 	assert.NoError(t, err)
 
+	// Incident creation may happen before plugins Access Monitoring Rule cache
+	// has been updated with new rule. Retry until the new cache picks up the rule.
+	require.EventuallyWithT(t, func(t *assert.CollectT) {
+		mu.Lock()
+		require.Contains(t, collectedNames, ruleName)
+		mu.UnLock()
+	}, 3*time.Second, time.Millisecond*100, "new access monitoring rule did not begin applying")
+
 	// Test execution: create an access request
 	req := s.CreateAccessRequest(ctx, integration.RequesterOSSUserName, nil)
 
 	// Validate the incident has been created in Pagerduty and its ID is stored
 	// in the plugin_data.
-	pluginData := s.checkPluginData(ctx, req.GetName(), func(data pagerduty.PluginData) bool {
+	hasIncidentID := s.checkPluginData(ctx, req.GetName(), func(data pagerduty.PluginData) bool {
```

Comment:
You can revert this name back to `pluginData`.

I was thinking the name of the closure that checks `data.IncidentID != ""` should be balled `hasIncidentID` when I originally made the suggestion, but that func is inlined so there's no need to name it at all.